src/setup/txt_joybinput.c

```diff
@@ -98,6 +99,9 @@ static void CanonicalizeButtons(void)
     int vbutton;
     int i;
 
+    static_assert(arrlen(all_joystick_buttons) == NUM_VIRTUAL_BUTTONS,
```

Comment:
Other arrays also just use NUM_VIRTUAL_BUTTONS to set their size, I don't know how nobody has though of that before. Seems like the best solution.